src/providers/abstract-provider.ts

```diff
@@ -677,6 +678,12 @@ export class AbstractProvider<C = FetchRequest> implements Provider {
 
     #options: Required<AbstractProviderOptions>;
 
+    _initFailed: boolean;
+
+    initResolvePromise: null | ((value: void) => void);
+    initRejectPromise: null | ((reason?: any) => void);
+    initPromise: Promise<void>;
```

Comment:
Should these be private or protected instance variables? They will default to public if not specified

Reply:
the resolve and reject should probably be protected. The initPromise is better to leave public bc the user could await for it to resolve if they want. But I guess they wouldn't really need to bc the sdk already does that for them. Idk, up to you.